src/pal/inc/pal.h

```diff
@@ -201,7 +201,13 @@ extern "C" {
 
 #else // _MSC_VER
 
+#if defined (_M_ARM) && !defined (DEBUG)
 #define UNALIGNED
+#define UNALIGNED_ARM     __attribute__((aligned(1))) //for -O2 and -O3
```

Comment:
> Can you just define the UNALIGNED to __attribute__((aligned(1))) instead of introducing the UNALIGNED_ARM?

There are too many "UNALIGNED" macros in CoreCLR. It looks like the trial to modify the existing UNALIGNED statements would be a tedious change to us too. 

* For more details:  my local branch (https://github.com/leemgs/coreclr/commit/fa3cc7892bef03c9ac5355e853a692dbd645bf77)
 , I have been working the next PR to remove `UNALIGNED_ARM` in the near future.